java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -27,6 +28,9 @@
 public class ScanOptions {
   private final Optional<String[]> columns;
   private final long batchSize;
+  private ByteBuffer projection;
+  private ByteBuffer filter;
```

Comment:
Should projection/filter be `final` if we have a builder for this object? We want the object to be immutable after creation I think.

Reply:
Changed